### ISSUE TITLE
Publish new version of the extension

### ISSUE THREAD

**Issue by jneira** [haskell/vscode-haskell#378]
* hls-1.1.0 has active the new configuration options but the extension is not
*  maybe we have to make some change to adapt it to hls-1.1.0 (new config options maybe?)
* we should test locally that master works with hls-1.1.0 (there is no ci job to do it right now). In windows, linux and macos if possible.

//cc @berberman @bubba

**Comment by berberman** [comment]
I think we should adapt to hls 1.1.0 asap, in case of more bug reports.

<details>
<summary>
The output of `cabal run haskell-language-server -- --vscode-extension-schema`:
</summary>

```json
{
    "haskell.plugin.brittany.globalOn": {
        "default": true,
        "description": "Enables brittany plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.class.codeActionsOn": {
        "default": true,
        "description": "Enables class code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.class.globalOn": {
        "default": true,
        "description": "Enables class plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.eval.codeLensOn": {
        "default": true,
        "description": "Enables eval code lenses",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.eval.globalOn": {
        "default": true,
        "description": "Enables eval plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.floskell.globalOn": {
        "default": true,
        "description": "Enables floskell plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.fourmolu.globalOn": {
        "default": true,
        "description": "Enables fourmolu plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-code-actions.codeActionsOn": {
        "default": true,
        "description": "Enables ghcide-code-actions code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-code-actions.globalOn": {
        "default": true,
        "description": "Enables ghcide-code-actions plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-completions.completionOn": {
        "default": true,
        "description": "Enables ghcide-completions completions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-completions.config.autoExtendOn": {
        "default": true,
        "markdownDescription": "Extends the import list automatically when completing a out-of-scope identifier",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-completions.config.snippetsOn": {
        "default": true,
        "markdownDescription": "Inserts snippets when using code completions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-completions.globalOn": {
        "default": true,
        "description": "Enables ghcide-completions plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-core.globalOn": {
        "default": true,
        "description": "Enables ghcide-core plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-hover-and-symbols.globalOn": {
        "default": true,
        "description": "Enables ghcide-hover-and-symbols plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-hover-and-symbols.hoverOn": {
        "default": true,
        "description": "Enables ghcide-hover-and-symbols hover",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-hover-and-symbols.symbolsOn": {
        "default": true,
        "description": "Enables ghcide-hover-and-symbols symbols",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-type-lenses.codeLensOn": {
        "default": true,
        "description": "Enables ghcide-type-lenses code lenses",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ghcide-type-lenses.config.mode": {
        "default": "always",
        "description": "Control how type lenses are shown",
        "enum": [
            "always",
            "diagnostics",
            "exported"
        ],
        "enumDescriptions": [
            "Always displays type lenses of global bindings",
            "Follows error messages produced by GHC about missing signatures",
            "Only display type lenses of exported global bindings"
        ],
        "scope": "resource",
        "type": "string"
    },
    "haskell.plugin.ghcide-type-lenses.globalOn": {
        "default": true,
        "description": "Enables ghcide-type-lenses plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.haddockComments.codeActionsOn": {
        "default": true,
        "description": "Enables haddockComments code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.haddockComments.globalOn": {
        "default": true,
        "description": "Enables haddockComments plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.hlint.codeActionsOn": {
        "default": true,
        "description": "Enables hlint code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.hlint.globalOn": {
        "default": true,
        "description": "Enables hlint plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.importLens.codeActionsOn": {
        "default": true,
        "description": "Enables importLens code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.importLens.codeLensOn": {
        "default": true,
        "description": "Enables importLens code lenses",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.importLens.globalOn": {
        "default": true,
        "description": "Enables importLens plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.moduleName.codeLensOn": {
        "default": true,
        "description": "Enables moduleName code lenses",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.moduleName.globalOn": {
        "default": true,
        "description": "Enables moduleName plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.ormolu.globalOn": {
        "default": true,
        "description": "Enables ormolu plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.pragmas.codeActionsOn": {
        "default": true,
        "description": "Enables pragmas code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.pragmas.completionOn": {
        "default": true,
        "description": "Enables pragmas completions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.pragmas.globalOn": {
        "default": true,
        "description": "Enables pragmas plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.retrie.codeActionsOn": {
        "default": true,
        "description": "Enables retrie code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.retrie.globalOn": {
        "default": true,
        "description": "Enables retrie plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.splice.codeActionsOn": {
        "default": true,
        "description": "Enables splice code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.splice.globalOn": {
        "default": true,
        "description": "Enables splice plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.stylish-haskell.globalOn": {
        "default": true,
        "description": "Enables stylish-haskell plugin",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.tactics.codeActionsOn": {
        "default": true,
        "description": "Enables tactics code actions",
        "scope": "resource",
        "type": "boolean"
    },
    "haskell.plugin.tactics.config.features": {
        "default": "",
        "markdownDescription": "Feature set used by Wingman",
        "scope": "resource",
        "type": "string"
    },
    "haskell.plugin.tactics.config.hole_severity": {
        "default": null,
        "description": "The severity to use when showing hole diagnostics. These are noisy, but some editors don't allow jumping to all severities.",
        "enum": [
            1,
            2,
            3,
            4,
            null
        ],
        "enumDescriptions": [
            "error",
            "hint",
            "info",
            "none",
            "warning"
        ],
        "scope": "resource",
        "type": "string"
    },
    "haskell.plugin.tactics.config.max_use_ctor_actions": {
        "default": 5,
        "markdownDescription": "Maximum number of `Use constructor <x>` code actions that can appear",
        "scope": "resource",
        "type": "integer"
    },
    "haskell.plugin.tactics.config.timeout_duration": {
        "default": 2,
        "markdownDescription": "The timeout for Wingman actions, in seconds",
        "scope": "resource",
        "type": "integer"
    },
    "haskell.plugin.tactics.globalOn": {
        "default": true,
        "description": "Enables tactics plugin",
        "scope": "resource",
        "type": "boolean"
    }
}
```

</details>

Obviously, it's not good enough to be dropped into `package.json`:
*  We don't need global switches of formatters, like `haskell.plugin.floskell.globalOn`
* `haskell.plugin.hlint.diagnosticsOn` is missing
* Top-level HLS options are missing

And here is a question: do we really need these `globalOn`s? For example, `haskell.plugin.class.globalOn` and `haskell.plugin.class.codeActionsOn` are for the same purpose, because class plugin only provides code actions. If one of both should be removed, which would be that one?

**Comment by jneira** [comment]
> And here is a question: do we really need these globalOns? For example, haskell.plugin.class.globalOn and haskell.plugin.class.codeActionsOn are for the same purpose, because class plugin only provides code actions. If one of both should be removed, which would be that one? And what about plugins which provide more than one capabilities?

Master already has support generic configuration, you can see the solution i adopted

https://github.com/haskell/vscode-haskell/blob/b14b3735d4582de485342cc8f5c51b2892caea5b/package.json#L155-L269

But maybe it has to be updated to add some config option added after i did that change

**Comment by berberman** [comment]
I saw it, but the pattern was not clear to me... Could you summarize the strategy we use? e.g. for single-function plugins with no extra custom config, we should drop `globalOn` and keep the specific `codeActionsOn` or `codeLensOn` .etc

**Comment by jneira** [comment]
Mmm, maybe i was not totally consistent, but the idea was:
* plugins with one capability: globalOn
* plugins with 2 capabilities:  one option per capability without globalOn (it is easy to turn off only 2 or three options)
